File: bindgen/data/audio/sound.lua
--[[
Copyright (c) 2021-present André Luiz Alvares
Nene is licensed under the Zlib license.
Please refer to the LICENSE file for details
SPDX-License-Identifier: Zlib
]]

local template = require 'bindgen.template'
local struct, field, fn = template.decls.struct,
                          template.decls.field,
                          template.decls.fn
local ast_reader = require 'bindgen.ast-reader'

local symbols = ast_reader.read_file('build/ast_dumps/audio/sound.txt', 'nene_Sound')

local structs = ast_reader.get_structs(symbols, 'nene')
local funcs = ast_reader.get_functions(symbols, 'nene_Sound')

return template.file(
  -- modname
  'Sound',
  -- links
  { template.build.link 'nene' },
  -- headers
  { template.build.header 'nene/audio/sound.h' },
  -- enums
  {},
  -- structs
  structs,
  -- funcs
  funcs,
  -- aliases
  {},
  -- global_vars
  {}
)

File: src/impl/audio/sound.c
/*
Copyright (c) 2021-present André Luiz Alvares
Nene is licensed under the Zlib license.
Please refer to the LICENSE file for details
SPDX-License-Identifier: Zlib
*/

#include "nene/impl/audio/sound.h"
#include "SDL_assert.h"

Mix_Chunk *nene_impl_Sound_get_raw(nene_Sound sound) {
	SDL_assert(sound.raw != NULL);
	return sound.raw;
}
